tests/unittests/test_handler/test_schema.py

```diff
@@ -447,4 +447,23 @@ def test_all_integration_test_cloud_config_schema(self):
         if errors:
             raise AssertionError(', '.join(errors))
 
+
+def _get_schema_doc_examples():
```

Comment:
What do you think about moving this definition into the class, so it's namespaced along with the test method that uses it?

Reply:
Yeah, I tried that already and I'm not sure it's possible without some weirdness. This decorator runs before the class has actually been defined, so I can only reference it if I have no `cls`, or `self` as the first argument, which is a really strange thing to do in a class. `@staticmethod` and `@classmethod` don't work either. Best to leave it outside I think.

Reply:
> which is a really strange thing to do in a class.

Yeah, it's definitely odd. I do like the idea of keeping all the bits related to a test (or set of related tests) under one top-level `class` definition, because it makes it easier to understand what's related to what. We have plenty of places in the code where we have 100s of lines of definitions of static test data defined at module-level, and then 100s of lines of test code which refer to that static test data; it's very difficult to infer what data is used where as a result. I _would_ also be comfortable with doing something a bit weird in our test code, if it made it easier to manage it (and provided we documented the pattern somewhere).

All that said, I think that fixtures are probably a more appropriate way of handling confusing test data reuse, and the test data usage here is not confusing, so I'm really suggesting a half-solution to a non-problem here.

As you were! ;-)